src/Microsoft.DotNet.Build.Tasks/PackageFiles/codeOptimization.targets

```diff
@@ -84,7 +85,7 @@
 
   <!-- We need the OptimizationData package in order to be able to optimize the assembly -->
   <Target Name="RestoreOptimizationDataPackage"
-          BeforeTargets="Sync"
+          BeforeTargets="Restore"
```

Comment:
Please just remove this we should avoid this "magic" hook-up because we will easily break it if we don't look for it.

Reply:
How would you suggest these targets get called?

Reply:
Actually you should be careful about changing this at all. If there are other repo's still hooking this up with Sync either changing it or removing will break those.

Reply:
>How would you suggest these targets get called?

It should be explicitly pulled in like I did in https://github.com/dotnet/corefx/pull/33076.